src/meta_memcache/base/base_cache_pool.py

```diff
@@ -0,0 +1,314 @@
+                for k in keys:
+                    self._write_failure_tracker.add_key(k)
+            raise
+        return results
+
+    def _conn_send_cmd(
+        self,
+        conn: MemcacheSocket,
+        command: MetaCommand,
+        key: Key,
+        value: Optional[bytes] = None,
+        flags: Optional[Set[Flag]] = None,
+        int_flags: Optional[Dict[IntFlag, int]] = None,
+        token_flags: Optional[Dict[TokenFlag, bytes]] = None,
+    ) -> None:
+        """
+        Execute command on a connection
+        """
+        cmd = self._build_cmd(
+            command,
+            key,
+            size=len(value) if value is not None else None,
+            flags=flags,
+            int_flags=int_flags,
+            token_flags=token_flags,
+        )
+        if value:
+            conn.sendall(cmd + value + ENDL)
+        else:
+            conn.sendall(cmd)
+
+    def _conn_recv_response(
+        self,
+        conn: MemcacheSocket,
+        flags: Optional[Set[Flag]] = None,
+    ) -> Union[Miss, Value, Success, NotStored, Conflict]:
+        """
+        Read response on a connection
+        """
+        if flags and Flag.NOREPLY in flags:
+            return Success(flags=set([Flag.NOREPLY]))
+        result = conn.get_response()
+        if isinstance(result, Value):
+            # TODO: Confirm this works for empty values!
+            data = conn.get_value(result.size)
+            if result.size > 0:
+                encoding_id = result.int_flags.get(IntFlag.CLIENT_FLAG, 0)
+                result.value = self._serializer.unserialize(data, encoding_id)
+        return result
+
+    def meta_multiget(
+        self,
+        keys: List[Key],
+        flags: Optional[Set[Flag]] = None,
+        int_flags: Optional[Dict[IntFlag, int]] = None,
+        token_flags: Optional[Dict[TokenFlag, bytes]] = None,
+    ) -> Dict[Key, Union[Miss, Value, Success]]:
+
+        results: Dict[Key, Union[Miss, Value, Success]] = {}
+        for key, result in self._exec_multi(
+            command=MetaCommand.META_GET,
+            keys=keys,
+            flags=flags,
+            int_flags=int_flags,
+            token_flags=token_flags,
+        ).items():
+            if not isinstance(result, (Miss, Value, Success)):
+                raise MemcacheError(
+                    f"Unexpected response for Meta Get command: {result}"
+                )
+            results[key] = result
+        return results
+
+    def meta_get(
+        self,
+        key: Key,
+        flags: Optional[Set[Flag]] = None,
+        int_flags: Optional[Dict[IntFlag, int]] = None,
+        token_flags: Optional[Dict[TokenFlag, bytes]] = None,
+    ) -> Union[Miss, Value, Success]:
+        result = self._exec(
+            command=MetaCommand.META_GET,
+            key=key,
+            flags=flags,
+            int_flags=int_flags,
+            token_flags=token_flags,
+        )
+        if not isinstance(result, (Miss, Value, Success)):
```

Comment:
Maybe 1 of 10 is failing, do we want to stop the whole multi command?

Reply:
This is not a multi command, I guess you wanted to comment on line `226`. If this happens, we do have a serious bug or glitch and we might be mixing data from other requests. We should not try and use any values and hard fail.